workflow/controller/controller.go

```diff
@@ -113,39 +105,107 @@ func (wfc *WorkflowController) Run(ctx context.Context) error {
 		return err
 	}
 
-	log.Info("Watch Workflow objects")
+	wfc.wfInformer = wfc.newWorkflowInformer()
+	wfc.podInformer = wfc.newPodInformer()
+	go wfc.wfInformer.Run(ctx.Done())
+	go wfc.podInformer.Run(ctx.Done())
 
-	// Watch Workflow objects
-	_, err = wfc.watchWorkflows(ctx)
-	if err != nil {
-		log.Errorf("Failed to register watch for Workflow resource: %v", err)
-		return err
+	// Wait for all involved caches to be synced, before processing items from the queue is started
```

Comment:
Why is this needed? Since the informers were just created, shouldn't their caches be empty anyway?

Reply:
I think that is the problem, is that the caches should not be empty, it should be synced and represent the current state of the cluster. By proceeding without it being synced, it will have an incorrect view of the world when the control loop begins. This code snippet was taken from their examples.

Reply:
oh.. ok. A comment here could help understand the need to sync state.